### ISSUE TITLE
Hard coded encryption algorithms prevent using this package in a FIPS environment

### ISSUE THREAD

**Issue by mcnallys** [Sustainsys/Saml2#812]
There are 2 different areas of the code preventing correct operation on a FIPS compliant machine.

"this implementation is not part of the Windows Platform FIPS validated cryptographic algorithms."

This stems from using the  RijndaelManaged symetricAlgorithm in CryptographyExtensions and some other issues that can be corrected in RSAEncryptedXML  using some of the code from

http://referencesource.microsoft.com/#System.Configuration/System/Configuration/FipsAwareEncryptedXml.cs,fad55741da0e4b5f

Would you be willing to accept a pull request to fix these issues?
When do you expect to release an update to nuget?

**Comment by AndersAbel** [comment]
Yes, I'd be willing to accept a PR to fix this. And I appreciate to have a discussion before you go ahead. One potential problematic place is the fix for when the cryptographic service provider associated with the certificate doesn't support SHA256: https://github.com/Sustainsys/Saml2/blob/c66a835af1a4acfd46051cf95a19a4093e15bf2f/Kentor.AuthServices/Internal/CryptographyExtensions.cs#L63 

For a Nuget release I'd say ASAP, but I've too much to do right now so can't promise a date.

If you want to get prioritized PR review and release coordination, that's something I can offer as part of my commercial services.

**Comment by agrady42** [comment]
I'm interested in this as well - 

For discussion - is there a way to set the algorithm used with configuration (even if it's something as simple as "useFipsCompliantAlgorithm=true" and document that this will cause problems with providers that don't support SHA256)?

For the rest - is there anything I can do to help with the PR?

**Comment by agrady42** [comment]
Open mouth, insert foot. Somehow I missed PR#813 just above my comment - thanks @mcnallys !

Just waiting on the release then.

**Comment by AndersAbel** [comment]
Fixed in v2.
